lib/Invocation.js

```diff
@@ -312,7 +320,16 @@ function scheduleNextRecurrence(rule, job, prevDate, endDate) {
 
   prevDate = (prevDate instanceof CronDate) ? prevDate : new CronDate();
 
-  const date = (rule instanceof RecurrenceRule) ? rule._nextInvocationDate(prevDate) : rule.next();
+  let date = null;
+
+  if (rule instanceof RecurrenceRule) {
```

Comment:
I would prefer to move this whole block to RRuleUtils and have something along the lines of
```
if (rule.isRRule) {
  const date = RRuleUtil.resolveNextInvocationDate(rule, prevDate)
}

Reply:
Not seeing what you mean here, this is essentially what you are suggesting? perhaps you could clarify

Reply:
Move
```
if (rule instanceof RecurrenceRule) {
    date = rule.nextInvocationDate(prevDate);
  } else if (rruleUtil.isRRuleObject(rule)) {
    date = rruleUtil.nextInvocationDate(rule, prevDate);
  } else {
    date = rule.next();
  }  
```
into `RRuleUtils` as e. g. `function resolveNextInvocationDate`.

replace this block here with

`date = rruleUtils.resolveNextInvocationDate()`

Basically same intent as stated before - encapsulate everything RRule-related inside RRule component.

Reply:
Thanks, I understand what you mean better know. Ill make some more changes tomorrow if I can and try to get that test passing :)

Reply:
Awesome, thanks!